core/include/tee/tee_fs.h

```diff
@@ -66,18 +67,19 @@ struct tee_fs_dirent {
 	char *d_name;
 };
 
+
 /*
- * tee_fs implemets a POSIX like secure file system
+ * tee_fs implemets a POSIX like secure file system with GP extension
  */
 struct tee_file_operations {
-	int (*open)(const char *file, int flags, ...);
+	int (*open)(TEE_Result *errno, const char *file, int flags, ...);
```

Comment:
Because the API has it own valid return value, e.g. open will return fd on success. It's better to return TEE_Result just like other sub-system, but we need to refactor tee_svc_storage.c and should re-design the tee_file_ops API.